src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3101,6 +3101,14 @@ async def __common_validation(self, verrors, schema, data, mode):
             for i in ('force_size', 'sparse', 'volsize', 'volblocksize'):
                 if i in data:
                     verrors.add(f'{schema}.{i}', 'This field is not valid for FILESYSTEM')
+
+            if 'special_small_block_size' in data and not (
+                data['special_small_block_size'] == 0 or 512 <= data['special_small_block_size'] <= 1000000
```

Comment:
I think this line will always return true for any value <= 1000000. The validation error says it can only be 0 and any value above 512?

Maybe write it like `size == 0 or size > 512 or size <= 1000000`

Reply:
also, maybe make a special_small_block_max_size variable and use that instead of a number?

Reply:
I am not sure if this `size > 512 <= 1000000` will work ?
We should just change the text

Reply:
>>> a
512
>>> a == 0 or a > 512 <= 1000000
False

Reply:
Ah that's a mistake on my end wrt text. However `512 <= size <= 1000000` is still a valid expression.

Reply:
Yep, I understand now that you've changed text 👍

Reply:
```
a = 1000001
a > 512 <= 1000000
```
This says true whereas we want it to be false

Reply:
I know, the verbiage confused me 😉 

My brain just likes writing things out explicitly (i.e. `a == 0 or a >= 512 and a <= 1000000`) but your way works too 👍